datafusion-examples/examples/parquet_encryption_with_kms.rs

```diff
@@ -0,0 +1,216 @@
+    ctx.register_parquet_encryption_factory(
+        ENCRYPTION_FACTORY_ID,
+        Arc::new(encryption_factory),
+    );
+
+    let tmpdir = TempDir::new()?;
+    write_encrypted(&ctx, &tmpdir).await?;
+    read_encrypted(&ctx, &tmpdir).await?;
+    Ok(())
+}
+
+/// Write an encrypted Parquet file
+async fn write_encrypted(ctx: &SessionContext, tmpdir: &TempDir) -> Result<()> {
+    let a: ArrayRef = Arc::new(StringArray::from(vec!["a", "b", "c", "d"]));
+    let b: ArrayRef = Arc::new(Int32Array::from(vec![1, 10, 10, 100]));
+    let batch = RecordBatch::try_from_iter(vec![("a", a), ("b", b)])?;
+
+    ctx.register_batch("test_data", batch)?;
+    let df = ctx.table("test_data").await?;
+
+    let mut parquet_options = TableParquetOptions::new();
+    // We specify that we want to use Parquet encryption by setting the identifier of the
+    // encryption factory to use.
+    //parquet_options.encryption.factory_id = ENCRYPTION_FACTORY_ID.to_owned();
+    parquet_options.global.encryption_factory_id = Some(ENCRYPTION_FACTORY_ID.to_owned());
+    // Our encryption factory requires specifying the master key identifier to
+    // use for encryption. To support arbitrary configuration options for different encryption factories,
+    // DataFusion could use a HashMap<String, String> field for encryption options.
+    //let encryption_config = EncryptionConfig::new("kf".to_owned());
+    //parquet_options.encryption.factory_options = encryption_config.to_config_map();
+
+    df.write_parquet(
+        tmpdir.path().to_str().unwrap(),
+        DataFrameWriteOptions::new(),
+        Some(parquet_options),
+    )
+    .await?;
+
+    println!("Encrypted Parquet written to {:?}", tmpdir.path());
+
+    Ok(())
+}
+
+/// Read from an encrypted Parquet file
+async fn read_encrypted(ctx: &SessionContext, tmpdir: &TempDir) -> Result<()> {
+    let mut parquet_options = TableParquetOptions::new();
+    // Specify the encryption factory to use for decrypting Parquet.
+    // In this example, we don't require any additional configuration options when reading
+    // as key identifiers are stored in the key metadata.
+    //parquet_options.encryption.factory_id = ENCRYPTION_FACTORY_ID.to_owned();
+    parquet_options.global.encryption_factory_id = Some(ENCRYPTION_FACTORY_ID.to_owned());
+
+    let file_format = ParquetFormat::default().with_options(parquet_options);
+    let listing_options = ListingOptions::new(Arc::new(file_format));
+
+    let file_name = std::fs::read_dir(tmpdir)?.next().unwrap()?;
+    let table_path = format!("file://{}", file_name.path().as_os_str().to_str().unwrap());
+
+    let _ = ctx
+        .register_listing_table(
+            "encrypted_parquet_table",
+            &table_path,
+            listing_options.clone(),
+            None,
+            None,
+        )
+        .await?;
+
+    let df = ctx.sql("SELECT * FROM encrypted_parquet_table").await?;
+    let plan = df.create_physical_plan().await?;
```

Comment:
While this works, I think it is probably more complex than needed and has a lot extra for setting up the config. I think it would be better to have a simpler example that just focuses on the KMS stuff. See the example and test I setup for more streamlined ways to read the table.